packages/math-input/src/components/keypad/operators-page/logarithms-buttons.tsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+
+import Keys from "../../../data/key-configs";
+import {PlaceHolderButtons, SecondaryKeypadButton} from "../keypad-page-items";
+
+type Props = {
+    onClickKey: (keyConfig: string) => void;
+    placeholder?: boolean;
+};
+
+export const Logarithms = ({
+    onClickKey,
+    placeholder,
+}: Props): React.ReactElement =>
+    placeholder ? (
+        <PlaceHolderButtons count={3} />
+    ) : (
+        <React.Fragment>
+            <SecondaryKeypadButton
+                keyConfig={Keys.LOG}
+                onClickKey={onClickKey}
+            />
+            <SecondaryKeypadButton
+                keyConfig={Keys.LOG_N}
+                onClickKey={onClickKey}
+            />
+            <SecondaryKeypadButton
+                keyConfig={Keys.LN}
+                onClickKey={onClickKey}
+            />
+        </React.Fragment>
+    );
```

Comment:
I'm not going to die on this hill I promise, but if we decide we want the placeholder in a subcomponent maybe we could make a PlaceholderRow that uses `children`:

```JSX
// possible definition
export const PlaceholderRow = ({
    children,
    placeholder,
}: Props): React.ReactElement =>
    placeholder ? <PlaceHolderButtons count={3} /> : children;
    
// possible usage
<PlaceholderRow placeholder={!this.props.logarithms}>
    <SecondaryKeypadButton
        keyConfig={Keys.LOG}
        onClickKey={onClickKey}
    />
    <SecondaryKeypadButton
        keyConfig={Keys.LOG_N}
        onClickKey={onClickKey}
    />
    <SecondaryKeypadButton
        keyConfig={Keys.LN}
        onClickKey={onClickKey}
    />
</PlaceholderRow>
```

That way:
1. We're not duplicating placeholder logic
2. We're not making a bunch of tiny components
3. We reduce our prop drilling

I know you and Jeremy talked about this, so if y'all decide to move forward on this I'll stop bringing it up. 😅

Reply:
Different button sets have different numbers of `PlaceholderButtons`, so we'd need that at least.

What do you mean by making a bunch of tiny components? The result of using `children` would be the same number of components, nested in a parent.

And this pattern, while great, would lead to a complete refactor of the KeyPad CSS. If `subgrid` was widely supported we could use that, but we would have to change our grid layout to accommodate this wrapper, then implement another grid within it.

Reply:
@handeyeco I went and implemented this but I still have to use it in the same place (within the PlaceholderButtons component) unless I just get rid of the component and put everything back into the parent. Otherwise we don't know the count (React can't see children through fragments). Or I restore the `count` attribute, which brings us right back to where we started.

Reply:
I'm going to land this as-is and we can do this as a future enhancement.